mage/algorithms/machine-learning-graph-analytics/temporal-graph-networks.md

```diff
@@ -0,0 +1,69 @@
+et al*. The last two methods offer a great possibility for inductive learning. But they haven't been specifically developed
+to handle different events occurring on graphs, such as **node features updates**, **node deletion**, **edge deletion**
+and so on. These events happen regularly on **real-world** examples such as **[Twitter network](https://twitter.com/memgraphmage)**, 
+where users update their profile, delete their profile or just unfollow another user.
+
+In their work, Rossi et al introduced to us [**Temporal graph networks**](https://arxiv.org/abs/2006.10637) which presents a great possibility to do graph machine
+learning on a stream of data, use-case occurring more often in recent years.
+
+
+There exist two main modules of **dynamic** graphs. Discrete-time dynamic graphs (DTDG) represent snapshots of static graphs taken at intervals in time. On the other hand continuous-time dynamic
+graphs (CTDG) are more general and can be represented as timed lists of events, which may include edge addition or deletion, node addition or deletion, and node or edge feature transformations.
+
+**Temporal graph networks** is a model of **CTDG**, which consists of time-stamped events. An **interaction** event is represented by temporal edge creation, and **node-wise** is represented by node creation or node feature update.
+
+
+## Materials
+* Firstly, read the **[original paper](https://arxiv.org/pdf/2006.10637.pdf)**.
+* Afterwards, you can also watch an explanation of **[temporal graph networks](https://www.youtube.com/watch?v=0tw66aTfWaI&t=1759s&ab_channel=TheAIEpiphany)**.
+* Be sure to check our explanation of **[node embeddings](https://memgraph.com/blog/introduction-to-node-embedding)** if you still have a problem with understanding what are node embeddings
+
+And finally, we prepared an example of  [**link prediction** on the Amazon dataset](https://hackmd.io/yy1mb9CVTYadmZz78iN5Jg).
+
+### Implementation
+
+[![Temporal Graph Networks](https://img.shields.io/badge/Node2Vec_Online-Implementation-FB6E00?style=for-the-badge&logo=github&logoColor=white)](https://github.com/memgraph/mage/blob/main/python/tgn.py)
+
+[![Temporal Graph Networks](https://img.shields.io/badge/Node2Vec_Online-Documentation-FCC624?style=for-the-badge&logo=python&logoColor=white)](/mage/query-modules/python/temporal-graph-networks)
+
+**Temporal graph networks** are implemented as part of the
+[**MAGE**](https://github.com/memgraph/mage) project. Be sure to check it out in
+the link above. :point_up:
+
+
+## Use cases
+
+[![Social
+networks](https://img.shields.io/badge/Social_networks-Application-8A477F?style=for-the-badge)](/use-cases/social-media.md)
+
+Temporal graph networks can be used to find communities in social networks like
+[Twitter](https://twitter.com), [Facebook](https://www.facebook.com/), and so
+on. To find communities, we can use **semi-supervised** learning, and from **graph structure** and **available** labels
```

Comment:
What are the use cases for supervised learning?

Reply:
Supervised is when we have all the data needed to do training epoch, but semi-supervised is better description here.

Reply:
I am not sure what are use cases for supervised, I think they are all part of semi-supervised as you don't have all data.